moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -32,132 +32,342 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  *********************************************************************/
 
-/* Author: Robert Haschke */
+/* Author: Robert Haschke, Mario Prats */
+
+// To run this benchmark, 'cd' to the build/moveit_core/robot_state directory and directly run the binary.
+
+#include <benchmark/benchmark.h>
+#include <kdl_parser/kdl_parser.hpp>
+#include <kdl/chainjnttojacsolver.hpp>
 #include <moveit/robot_model/robot_model.h>
 #include <moveit/robot_state/robot_state.h>
 #include <moveit/utils/robot_model_test_utils.h>
-#include <eigen_stl_containers/eigen_stl_containers.h>
-#include <chrono>
-#include <gtest/gtest.h>
+#include <random_numbers/random_numbers.h>
 
-// Helper class to measure time within a scoped block and output the result
-class ScopedTimer
-{
-  const char* const msg_;
-  double* const gold_standard_;
-  const std::chrono::time_point<std::chrono::steady_clock> start_;
+// Robot and planning group for benchmarks.
+constexpr char PANDA_TEST_ROBOT[] = "panda";
+constexpr char PANDA_TEST_GROUP[] = "panda_arm";
+constexpr char PR2_TEST_ROBOT[] = "pr2";
+constexpr char PR2_TIP_LINK[] = "r_wrist_roll_link";
 
-public:
-  // if gold_standard is provided, a relative increase/decrease is shown too
-  ScopedTimer(const char* msg = "", double* gold_standard = nullptr)
-    : msg_(msg), gold_standard_(gold_standard), start_(std::chrono::steady_clock::now())
+// Number of iterations to use in matrix multiplication / inversion benchmarks.
+constexpr int MATRIX_OPS_N_ITERATIONS = 1e7;
+
+static void multiplyAffineTimesMatrix(benchmark::State& st)
+{
+  int n_iters = st.range(0);
+  Eigen::Isometry3d isometry = Eigen::Translation3d(1, 2, 3) *
+                               Eigen::AngleAxisd(0.13 * M_PI, Eigen::Vector3d::UnitX()) *
```

Comment:
Can you add an explanation for the magic numbers 0.13, 0.29 and 0.42?

Reply:
Moved this into a helper function and added a comment saying these are arbitrary test values.